cypress/e2e/with-users/machines/details.spec.ts

```diff
@@ -24,4 +24,37 @@ context("Machine details", () => {
     cy.findAllByRole("columnheader", { name: /IP/i }).first().should("exist");
     cy.findByRole("columnheader", { name: /subnet/i }).should("not.exist");
   });
+
+  it("displays machine commissioning details", () => {
+    cy.findByRole("grid").within(() => {
+      cy.findAllByRole("gridcell", { name: /FQDN/i })
+        .first()
+        .within(() => cy.findByRole("link").click());
+    });
+    cy.waitForPageToLoad();
+    cy.get("[data-testid='section-header-buttons']").within(() =>
+      cy
+        .findByRole("button", {
+          name: /Take action/i,
+        })
+        .click()
+    );
+
+    // abort commissioning
```

Comment:
I think this uncovers a bit of an issue with these tests. I presume we need to abort commissioning here because this test expects `machines/add.spec.ts` to have run first and so we assume that the machine that has been added there will still be commissioning.

I think instead we should set up a machine in the `beforeEach` (or potentially `beforeAll` if we think we can safely use the same machine for all tests) of this file to remove the dependent (and leaky) tests as well as removing this race condition.

This will slow these tests down, but make them more reliable.

Reply:
Agreed, testing interactions that involve state updates in the MAAS back-end are tricky. I'm not sure about `beforeEach`, I'm concerned that if we keep going at this pace, we'll end up with the same problem that we had in the past - our test suite taking ages to complete. I'll look into this.